serving/src/main/java/ai/djl/serving/http/KServeRequestHandler.java

```diff
@@ -174,4 +392,124 @@ private void onException(Exception ex, ChannelHandlerContext ctx) {
 
         NettyUtils.sendJsonResponse(ctx, content, status);
     }
+
+    void onModelException(Throwable t, ChannelHandlerContext ctx) {
+        HttpResponseStatus status;
+        if (t instanceof TranslateException) {
+            status = HttpResponseStatus.BAD_REQUEST;
+        } else if (t instanceof WlmException) {
+            logger.warn(t.getMessage(), t);
+            status = HttpResponseStatus.SERVICE_UNAVAILABLE;
+        } else {
+            logger.warn("Unexpected error", t);
+            status = HttpResponseStatus.INTERNAL_SERVER_ERROR;
+        }
+        if (ctx != null) {
+            NettyUtils.sendError(ctx, status, t);
+        }
+    }
+}
+
+class RequestInput {
+    private String name;
+    private List<Long> shape;
+    private String datatype;
+    // TODO: accept all types of data
+    private List<Double> data;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public List<Long> getShape() {
+        return shape;
+    }
+
+    public void setShape(List<Long> shape) {
+        this.shape = shape;
+    }
+
+    public String getDatatype() {
+        return datatype;
+    }
+
+    public void setDatatype(String datatype) {
+        this.datatype = datatype;
+    }
+
+    public List<Double> getData() {
+        return data;
+    }
+
+    public void setData(List<Double> data) {
+        this.data = data;
+    }
+
+    public String toString() {
+        return "Input [ name: "
+                + name
+                + ", datatype: "
+                + datatype
+                + ", shape: "
+                + shape
+                + ", data "
+                + data
+                + " ]";
+    }
+}
+
+class RequestOutput {
+    private String name;
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String toString() {
+        return "Output [ name: " + name + " ]";
+    }
+}
+
+class InferenceResponse {
+    private ArrayList<ResponseOutput> outputs;
+    private String id;
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public void setList(ArrayList<ResponseOutput> responseOutput) {
+        this.outputs = responseOutput;
+    }
+}
+
+class ResponseOutput {
```

Comment:
ResponseOutput and RequestInput have the same format. You can use single class for this.